### ISSUE TITLE
"perfect day" is awarded if all Dailies are unticked Grey Dailies, or there are no Dailies

### ISSUE THREAD

**Issue by Alys** [HabitRPG/habitica#2565]
If you have no Dailies or all your Dailies are greyed out because none are set to be actioned that day, then you get a "perfect day" achievement, even if you didn't tick off any Dailies at all.

You also get a Perfect Day if you use Stealth to avoid all Dailies.

There's a proposed fix at https://github.com/HabitRPG/habitrpg/issues/4535 but it hasn't been tested and it's very old now. 
##

**Comment by Nurylon** [comment]
Hopefully not make it so that having /any/ dailies (as apposed to all) that are greyed out every day make one unable to get the achievement, since that can be non-malicious. I have some things in dailies like that that are there because I do them only once a day if I do them, but I don't feel they should hurt me if I can't do them. For instance, I have getting dressed to go out as a daily, but I have very few clothes so I can't afford to dirty one of my only two pants on days where I'm not going anywhere since it'd use too much laundry. And things I like to do every day, but are ultimanteily based on cooperation with other people (like family meetings or greeting my partner in his morning) are greyed out, too.

**Comment by Alys** [comment]
Hi Nuryion. That's not something you need to be concerned about. With the current code, you'll get your perfect day achievement if you have no uncompleted dailies (i.e., you either actioned all your dailies, or they were all greyed-out [that's the bug], or some were greyed-out and you actioned all of the others). With the proposed fix (or indeed any fix for this bug), you'll still get your perfect day achievement when you action ALL your NON-GREY dailies. I.e., it's still okay to have greyed-out dailies as long as you action ALL of the others.

The only change this fix will make is that you don't get the achievement when you haven't done any dailies at all because they were all greyed-out.

**Comment by Alys** [comment]
https://github.com/HabitRPG/habitrpg/issues/4535#issue-54718874

**Comment by librarianmage** [comment]
What is the progress on this?

**Comment by Alys** [comment]
Let's put this one on hold for now. We'd need to modify cron for this,
which is best done after version 3 of the API has been released (in
development now).

**Comment by librarianmage** [comment]
@Alys that makes sense (much to do after v3 is released)

**Comment by khipkin** [comment]
I think the [wait] label can be removed from this now.

**Comment by khipkin** [comment]
I can work on this!

**Comment by paglias** [comment]
@khipkin feel free to go ahead :)
